### PR TITLE
armpl-gcc: add macOS support and map rocky9 to RHEL-9

### PR DESCRIPTION
**Pull request by annop-w**

### CODE REVIEW

File: var/spack/repos/builtin/packages/armpl-gcc/package.py [hunk 1]
```diff
@@ -177,20 +181,26 @@
 
 
 def get_os(ver):
-    spack_os = spack.platforms.host().default_os
+    platform = spack.platforms.host()
+    if platform.name == "darwin":
+        return "darwin"
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/armpl-gcc/package.py]: The name of this function is kind of misleading now. `darwin` is the platform, not the OS.

File: var/spack/repos/builtin/packages/armpl-gcc/package.py [hunk 2]
```diff
@@ -177,20 +181,26 @@
...
     armpl_version = version.split("_")[0]
     armpl_version_dashed = armpl_version.replace(".", "-")
-    gcc_version = version.split("_")[1]
+    compiler_version = version.split("_", 1)[1]
     os = get_os(armpl_version)
-    filename = "arm-performance-libraries_" + armpl_version + "_" + os + "_" + gcc_version + ".tar"
+    if os == "darwin":
+        return "{}{}/armpl_{}_{}.dmg".format(
+            base_url, armpl_version_dashed, armpl_version, compiler_version
+        )
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/armpl-gcc/package.py]: Can we use f-strings here?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/armpl-gcc/package.py (hunk 1 above)
Comment (annop-w) [reply var/spack/repos/builtin/packages/armpl-gcc/package.py]: Done.

File: var/spack/repos/builtin/packages/armpl-gcc/package.py (hunk 2 above)
Comment (annop-w) [reply var/spack/repos/builtin/packages/armpl-gcc/package.py]: Done.

**Comment by annop-w** [comment]
@alalazo Could you please give this a quick look ? Thank  you.

**Comment by annop-w** [comment]
@alalazo Do you have any more comment on this one ? We'll have an update for the next release pretty soon. So, it would be great to have this one in.

**Pull request closed without merging**